### ISSUE TITLE
Bazecor is consuming a lot of CPU when idle

### ISSUE THREAD

**Issue by wAuner** [Dygmalab/Bazecor#234]
**Describe the bug**
macOS 11.4 Apple Silicon
Bazecor is running at ~40% usage if it is idle in the background. Needless to say that that's pretty bad.
<img width="767" alt="Screenshot 2021-06-11 at 11 54 56" src="https://user-images.githubusercontent.com/18383180/121668937-175ba600-caac-11eb-8344-35fa505f9ca3.png">

As soon as I connect to the keyboard in Bazecor, the resource consumption explodes:

<img width="797" alt="Screenshot 2021-06-11 at 11 58 29" src="https://user-images.githubusercontent.com/18383180/121669171-57228d80-caac-11eb-903f-d1848d385178.png">

**Comment by GazHank** [comment]
@wAuner are you running on an M1?  If so have you performed a full build/ rebuild or are you running the Intel software version in compatability mode?

**Comment by wAuner** [comment]
This is on Apple Silicon (M1) and it is the release version running translated for arm (Rosetta 2). The `kind` column stating Intel shows that. Otherwise it would say Apple.

**Comment by GazHank** [comment]
@wAuner would you be able to try recompiling from the source code?  (I can try to provide some help/ instructions for the process if that would be useful)

In addition to high processor load, I fear you may also hit issues when trying to upgrade the firmware in future if using the intel version of the software.   

Given the different way in which apple have setup the USB hardware/firmware for the new devices quite a few of the packages we rely upon will need to be recompiled rather than run in compatibility mode.  E.g. https://github.com/serialport/node-serialport/issues/2243

Unfortunately Apple haven't enabled cross compile from another platform, and GitHub haven't yet release the new M1 machines, so you would need to recompile for M1 processor locally

**Comment by wAuner** [comment]
I could try, if there isn’t too much setup and dependencies involved. Could you provide an instruction? 
I don’t quite understand why it shouldn’t be possible to cross compile from any other platform, especially not macOS x86. Unless you’re using some Apple Frameworks such as DriverKit you should even be able to compile from Linux.

**Comment by GazHank** [comment]
@wAuner you’re absolutely right, it’s DriverKit/ IOKit which are used for the low level USB interaction, hence the problem.

From what I’ve read, it is the way that the USB stack has been engineered for the new architecture that is one the biggest changes; and the largest headache for the people porting Linux to the new architecture.  I don’t know why apple haven’t focussed on enabling cross compile (after all you can compile for the iPhone from an intel machine), but this is affecting lots of projects from what I have seen.

I’ve written some build instruction in response to https://github.com/Dygmalab/Bazecor/issues/45

While the instructions were for Linux/ Windows the main steps should be the same, the main difference would be the build tools you will need, Xcode of course, but I’m not sure if anything else is needed for Mac.  Of course if you can help clarify or refine the steps for Mac then that would be fab!

While the yarn rebuild step wasn’t required on Linux, I certainly expect it to be needed on the new Mac; and it is that step where I expect you might hit some errors during the compilations.

Do please let me know if you hit any problems with the instructions or hit any errors with the build and I’ll try to help

Gaz

**Comment by javierguzman** [comment]
Hello there! We have upgraded the internal build system so now we use Electron Forge. I think performance should be fairly better now. Please, let us know if this continues in the latest version (prerelease).

**Comment by alexpargon** [comment]
Tested this several times in M1 & M2 hardware and now the CPU usage goes to zero when not actively using the App, either in the background or simply not interacting with it.
